src/app/base/components/AppSideNavigation/AppSideNavItem/AppSideNavItem.tsx

```diff
@@ -0,0 +1,50 @@
+import { Icon } from "@canonical/react-components";
+import { Link } from "react-router-dom-v5-compat";
+
+import type { NavItem } from "../types";
+import { isSelected } from "../utils";
+
+import { useId } from "app/base/hooks/base";
+
+type Props = {
+  navLink: NavItem;
+  icon?: string;
+  hasWarningIcon?: boolean;
+  path: string;
+};
+
+export const AppSideNavItem = ({
+  navLink,
+  icon,
+  hasWarningIcon,
+  path,
+}: Props): JSX.Element => {
+  const id = useId();
+  return (
+    <li
+      aria-labelledby={`${navLink.label}-${id}`}
+      className={`l-navigation__item${
+        isSelected(path, navLink) ? " is-selected" : ""
+      }`}
+    >
+      <Link
+        aria-current={isSelected(path, navLink) ? "page" : undefined}
+        className="l-navigation__link"
+        id={`${navLink.label}-${id}`}
+        to={navLink.url}
+      >
+        {hasWarningIcon ? (
```

Comment:
This could be simplified and merged with the `icon` prop.

Reply:
Good point. I've adjusted the icon prop to take a whole icon component in this case, since the warning icon has some additional props that it needs for accessibility and testing.